src/features/security-cloud-game/__tests__/DragCloud.test.js

```diff
@@ -1,20 +1,114 @@
 import React from 'react';
+import Draggable from 'react-draggable';
 
-import DragCloud from '../DragCloud';
+import { DragCloudBase as DragCloud } from '../DragCloud';
+
+const intl = {
+    formatMessage: message => message.defaultMessage,
+};
 
 describe('features/security-cloud-game/DragCloud', () => {
     test('should correctly render', () => {
-        const component = shallow(<DragCloud cloudSize={100} position={{ x: 10, y: 20 }} />);
-
-        expect(component.hasClass('drag-cloud')).toBe(true);
-        expect(component.prop('style')).toEqual({
-            top: '20px',
-            left: '10px',
-        });
-        expect(component.find('IconCloud').length).toEqual(1);
-        expect(component.find('IconCloud').prop('height')).toEqual(100);
-        expect(component.find('IconCloud').prop('width')).toEqual(100);
-        expect(component.find('IconCloud').prop('filter')).toBeDefined();
-        expect(component.find('IconCloud').prop('filter').id).toEqual('drop-shadow');
+        const wrapper = shallow(
+            <DragCloud
+                gameBoardSize={{ height: 500, width: 500 }}
+                cloudSize={100}
+                gridTrackSize={20}
+                intl={intl}
+                position={{ x: 10, y: 20 }}
+            />,
+        );
+
+        const draggable = wrapper.find(Draggable);
+        expect(draggable.prop('position')).toEqual({ x: 10, y: 20 });
+        expect(draggable.find('.bdl-DragCloud').length).toEqual(1);
```

Comment:
instead of `length`, could we use `exists`?
```suggestion
        expect(draggable.exists('.bdl-DragCloud')).toBe(true);
```

Reply:
Yes, I changed this line and all other places accordingly where `length` were used.